addons/spreadsheet/models/spreadsheet_mixin.py

```diff
@@ -21,6 +23,45 @@ class SpreadsheetMixin(models.AbstractModel):
     spreadsheet_data = fields.Text(compute='_compute_spreadsheet_data', inverse='_inverse_spreadsheet_data')
     thumbnail = fields.Binary()
 
+    @api.constrains("spreadsheet_binary_data")
+    def _check_spreadsheet_data(self):
+        for spreadsheet in self.filtered("spreadsheet_binary_data"):
+            try:
+                data = json.loads(base64.b64decode(spreadsheet.spreadsheet_binary_data).decode())
+            except (json.JSONDecodeError, UnicodeDecodeError):
+                raise ValidationError(_("Uh-oh! Looks like the spreadsheet data file isn't quite right."))
```

Comment:
🔥 🔥 🔥

Reply:
I must say I take no credit for this. It's based on an original artistic direction by a guy whose initials are F.P. and executed by Mr. GPT (firstname is Chat :cat2: )

Reply:
"Mister ORM, I don't feel so good"

More seriously, I would say (in this specific case) that the data file is corrupted.